cmd/thriftbreak/main_test.go

```diff
@@ -60,30 +85,33 @@ func TestThriftBreakIntegration(t *testing.T) {
 		"somefile.go": `service Qux{}`,
 	}
 	remove := []string{"test/d.thrift"}
-	breaktest.CreateRepoAndCommit(t, tmpDir, from, to, remove)
 
-	f, err := ioutil.TempFile(tmpDir, "stdout")
-	require.NoError(t, err, "create temporary file")
-	defer func(oldStdout *os.File) {
-		assert.NoError(t, f.Close())
-		os.Stdout = oldStdout
-	}(os.Stdout)
-	os.Stdout = f
+	for _, tt := range tests {
+		t.Run(tt.desc, func(t *testing.T) {
+			tmpDir := t.TempDir()
+			breaktest.CreateRepoAndCommit(t, tmpDir, from, to, remove)
+
+			f, err := ioutil.TempFile(tmpDir, "stdout")
+			require.NoError(t, err, "create temporary file")
+			defer func(oldStdout *os.File) {
```

Comment:
stdout hijacking is probably unnecessary now that you have jsonOutput and readableOutput but I assume you'll be doing that in a follow-up PR?

Reply:
I think the hijacking is still needed if while the code is still calling `run()` because run does not yet parameterize stdout. @shirchen plans to fix that in GO-892.

The unit tests for json/readableOutput don't need stdout hijacking, though, that's true.